### ISSUE TITLE
`tabIndex={0}` dont seems to set the `accessible` props on IOS

### ISSUE THREAD

**Issue by t1gu1** [tamagui/tamagui#2698]
### Current Behavior

If I want to activate the accessibility on an IOS device, in the following code (For a switch) the props `accessible` is required to the switch to be even focusable be the screenreader on IOS.

Also if I remember correctly, this is the same behaviors for buttons and link without text. (Example just an with an icon)

![image](https://github.com/tamagui/tamagui/assets/12479055/e868a51f-25e2-4c4c-8786-2bc3187d4d6b)

![image](https://github.com/tamagui/tamagui/assets/12479055/fc3b17b7-0ece-4a73-947d-a4627920cdae)

Also, when the switch is check, it always said the invert only at the moment we click and it annouce the change.

```tsx
<Switch
        size="$4"
        checked={currentTheme === "dark"}
        onCheckedChange={() =>
          setCurrentTheme(currentTheme === "dark" ? "light" : "dark")
        }
        bg="darkgray"
        cursor="pointer"
        aria-label="Active dark theme"
        accessible // <- This is required for a11y
        nativeProps // <- A nativeProps is required (Event empty do the job)
      >
        <Switch.Thumb animation="bouncy" />
      </Switch>
```

### Expected Behavior

From what the types seems to say, it should be `tabIndex={0}` i guess.
But it's the type of component that should be accessible by default I think?

For the moment the problem is it doesnt add the `accessible` props for IOS and i guess the `importantForAccessibility` on android. (Not tested on Android, so im not sure)

```tsx
<Switch
        size="$4"
        checked={currentTheme === "dark"}
        onCheckedChange={() =>
          setCurrentTheme(currentTheme === "dark" ? "light" : "dark")
        }
        bg="darkgray"
        cursor="pointer"
        aria-label="Active dark theme"
        
        tabIndex={0} // <-- I guess?
      >
        <Switch.Thumb animation="bouncy" />
      </Switch>
```

### Tamagui Version

```markdown
1.100.0
```

### Platform (Web, iOS, Android)

```markdown
I only test it on IOS
```

### Reproduction

```markdown
<Switch
        size="$4"
        p={0} // <-- Style is broken without p={0} on web
        checked={currentTheme === "dark"}
        onCheckedChange={() =>
          setCurrentTheme(currentTheme === "dark" ? "light" : "dark")
        }
        bg="darkgray"
        cursor="pointer"
        aria-label="Active dark theme"
        accessible // <- This is required
      >
             <Switch.Thumb animation="bouncy" />
</Switch>
```

### System Info

```markdown
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 2.29 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 4.0.2 - ~/.nvm/versions/node/v18.16.0/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
    bun: 1.0.2 - ~/.bun/bin/bun
    Watchman: 2024.05.06.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 125.0.6422.142
    Edge: 125.0.2535.85
    Safari: 17.5
  npmPackages:
    @babel/core: ^7.24.5 => 7.24.5 
    @expo/metro-runtime: ~3.2.1 => 3.2.1 
    @react-native-async-storage/async-storage: ^1.23.1 => 1.23.1 
    @react-navigation/drawer: ^6.6.15 => 6.6.15 
    @react-navigation/native: 6.1.17 => 6.1.17 
    @tamagui/config: ^1.100.0 => 1.100.0 
    @tamagui/lucide-icons: ^1.100.0 => 1.100.0 
    @tamagui/metro-plugin: ^1.100.0 => 1.100.0 
    @tamagui/toast: ^1.100.0 => 1.100.0 
    @types/react: ~18.2.79 => 18.2.79 
    @types/react-native: ^0.73.0 => 0.73.0 
    babel-preset-expo: ^11.0.6 => 11.0.6 
    burnt: ^0.12.2 => 0.12.2 
    expo: ~51.0.8 => 51.0.8 
    expo-font: ~12.0.5 => 12.0.5 
    expo-linking: ~6.3.1 => 6.3.1 
    expo-localization: ~15.0.3 => 15.0.3 
    expo-router: ~3.5.14 => 3.5.14 
    expo-splash-screen: ~0.27.4 => 0.27.4 
    expo-status-bar: ^1.12.1 => 1.12.1 
    expo-system-ui: ~3.0.4 => 3.0.4 
    expo-video: ^1.1.10 => 1.1.10 
    expo-web-browser: ~13.0.3 => 13.0.3 
    i18next: ^23.11.5 => 23.11.5 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.3.1 
    react-i18next: ^14.1.2 => 14.1.2 
    react-native: 0.74.1 => 0.74.1 
    react-native-reanimated: ~3.10.1 => 3.10.1 
    react-native-safe-area-context: 4.10.1 => 4.10.1 
    react-native-screens: ~3.31.1 => 3.31.1 
    react-native-svg: ^15.2.0 => 15.3.0 
    react-native-web: ^0.19.12 => 0.19.12 
    tamagui: ^1.100.0 => 1.100.0 
    typescript: ^5.3.3 => 5.4.5 
    zustand: ^4.5.2 => 4.5.2
```

**Comment by natew** [comment]
@bidah this one could be a decent first one, seems like it's mostly just adjusting some props we set in `getSplitStyles` or `createComponent` somewhere

**Comment by bidah** [comment]
Picking this up now

**Comment by natew** [comment]
Fixed in 6e1eab5428b2a6090806f053b14f56bf717fa62f and 5d32a6c54ed8d3a5f11edc506484bfa5dd446727
